### PR TITLE
allow distinct(true) query expresion

### PR DESCRIPTION
**Pull request by jxs**

### CODE REVIEW

File: lib/ecto/query.ex [hunk 1]
```diff
@@ -436,6 +436,10 @@ defmodule Ecto.Query do
     Distinct.build(query, binding, expr, __CALLER__)
   end
 
+  defmacro distinct(query, expr) do
```
Comment (josevalim) [inline lib/ecto/query.ex]: We need to add docs. Even if it will be just "Adds a distinct clause. See `distinct/3`."

Comment (jxs) [inline lib/ecto/query.ex]: yeah of course, going to update :)

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by josevalim** [comment]
Actually, I just realized we could do `binding \\ []` in the first definition. I will fix it locally :D

**Pull request merged**